common/src/main/java/org/openmrs/analytics/OpenmrsUtil.java

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.openmrs.analytics;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.client.api.IClientInterceptor;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.client.interceptor.BasicAuthInterceptor;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenmrsUtil {
```

Comment:
I separated out the util functions for the source and target to make things a bit cleaner and unit-testing easier. Since both use `fhirContext` and it's expensive to create, I pulled it out.

Reply:
Looks good; thanks.
For future, does it make sense to use HAPI FHIR client to push resources to the target store too?

Reply:
yes - I'm using the client to push to a HAPI JPA server for the iSantePlus work, and was planning on a PR for it. 

Some target repositories might have their own clients that could be better to use in those cases, but we can definitely use the HAPI client as a default for generic communication, replacing the HttpClient